server/ladder_service.py

```diff
@@ -335,6 +336,8 @@ async def start_game(
         team2: List[Player],
         queue: MatchmakerQueue
     ) -> None:
+        assert len(team1) == len(team2)
```

Comment:
How is this handled if it fails? (just asking since it seems to be somewhat unrelated)

Reply:
If it does fail, I think there would be a warning about an unhandled exception in a coroutine, but I only put it there because it is a precondition of the function. It's basically a way of saying to the programmer "never call this function with teams of different sizes", plus if the programmer ever does that, then hopefully it will cause a unit test to fail. 

I would like to actually run the server with python's `-O` option which removes all assert statements at some point, except that I do know there are 1 or 2 assert statements in the code (which were there when I took over) that do occasionally fail and show up in the prod server logs. So those would need to be fixed first.